rust/agents/relayer/src/msg/gelato_submitter.rs

```diff
@@ -1,98 +1,231 @@
-use abacus_base::{chains::GelatoConf, InboxContracts};
-use abacus_core::db::AbacusDB;
-use tokio::task::JoinHandle;
-
-use eyre::Result;
+use abacus_base::CoreMetrics;
+use abacus_core::{db::AbacusDB, Encode, Signers};
+use abacus_ethereum::validator_manager::INBOXVALIDATORMANAGER_ABI as ivm_abi;
+use ethers::abi::Token;
+use ethers::types::{Address, U256};
+use ethers_contract::BaseContract;
+use eyre::{bail, Result};
+use gelato::chains::Chain;
+use gelato::fwd_req_call::{ForwardRequestArgs, PaymentType, NATIVE_FEE_TOKEN_ADDRESS};
+use prometheus::{Histogram, IntCounter, IntGauge};
 use tokio::sync::mpsc;
+use tokio::time::{sleep, Duration};
+use tokio::{sync::mpsc::error::TryRecvError, task::JoinHandle};
 use tracing::{info_span, instrument::Instrumented, Instrument};
 
 use super::SubmitMessageArgs;
 
-// TODO(webbhorn): Metrics data.
-// TODO(webbhorn): Pull in ForwardRequestOp logic from prior branch.
+const DEFAULT_MAX_FEE: u32 = 1_000_000_000;
```

Comment:
I would really like @tkporter to weigh in on this. Kinda scares me but I also appreciate the desire to not have a full mapping here.

Reply:
Definitely will wait.

Reply:
How about we ask Gelato if they care about max fees at all on testnets? by "care" I mean if they will ever not process a testnet forwardrequest because the max fee is too low. My understanding is they don't charge on testnet so I'd be surprised if they cared. But if they do care and they don't charge on testnet, let's just pick something high like 1e18 (1 ether) ?

But we should definitely put a comment here indicating that it's temporary and only expected to be ran for testnet